// @clerk/shared/dist/chunk-PBZPBRK6.mjs
import {
  pathToRegexp
} from "./chunk-JJHTUJGL.mjs";

// src/pathMatcher.ts
var MalformedURLError = class extends Error {
  constructor(pathname, cause) {
    super(`Malformed encoding in URL path: ${pathname}`);
    this.statusCode = 400;
    this.name = "MalformedURLError";
    this.cause = cause;
  }
};
function isMalformedURLError(e) {
  return e instanceof Error && e.name === "MalformedURLError";
}
var precomputePathRegex = (patterns) => {
  return patterns.map((pattern) => pattern instanceof RegExp ? pattern : pathToRegexp(pattern));
};
var normalizePath = (pathname) => {
  try {
    pathname = decodeURI(pathname);
  } catch (e) {
    throw new MalformedURLError(pathname, e);
  }
  return pathname.replace(/\/\/+/g, "/");
};
var createPathMatcher = (patterns) => {
  const routePatterns = [patterns || ""].flat().filter(Boolean);
  const matchers = precomputePathRegex(routePatterns);
  return (pathname) => matchers.some((matcher) => matcher.test(normalizePath(pathname)));
};

export {
  MalformedURLError,
  isMalformedURLError,
  normalizePath,
  createPathMatcher
};
